### DISCUSSION TITLE
Hard to answer curb height quest

### DISCUSSION THREAD

**Discussion by smichel17** [streetcomplete/StreetComplete#3344] [Feedback]
Here's two recent examples where I wasn't sure how to answer.

https://www.openstreetmap.org/note/2845421
https://www.openstreetmap.org/note/2866316

<img height="500" src="https://user-images.githubusercontent.com/6563664/134583120-e9b4756a-97aa-4efa-bf29-3f3fd68c8288.png"> <img height="500" src="https://user-images.githubusercontent.com/6563664/134583194-5a58d509-8fc5-4adf-9411-0b7dc6e34862.png">

The current form looks like this: <img height="600" src="https://user-images.githubusercontent.com/6563664/134583797-d22d84ec-1343-4911-ab9b-9f90c6a1715e.png"> (v34.1 from F-Droid)

- It looks kind of like a curb ramp, except the sidewalk itself slopes down, not a separate piece.

- It looks kind of like "same level" or "none at all", except that it is at the end of the sidewalk, not the side.

Additionally, it's sometimes hard to tell which curb exactly is being asked about: the border between the sidewalk and a nearby street (along the whole way), or the situation for getting on to the sidewalk from the street? Here's an example of where this confused me (for the curb tactile paving quest): https://www.openstreetmap.org/note/2744876 

![sc-curb-where](https://user-images.githubusercontent.com/6563664/134584757-56b0f4cf-352d-47b1-b1a2-88618a52412c.jpg)

**Comment by westnordost** [comment]
Well, I'd say it is an error in mapping, so it's right that you left a note:
![image](https://user-images.githubusercontent.com/4661658/134589468-a379d6bd-661e-4ce2-95b5-db40fcc42391.png)

**Comment by westnordost** [comment]
Regarding which answer to select: Well, there is clearly a change in paving, so "same level".

**Comment by smichel17** [comment]
Hm, I do not think this addresses the issue, which is that I've previously tagged some similar things as "curb ramp" due to this confusion. I will get more pictures of "on the edge" curbs to demonstrate more clearly.

**Comment by smichel17** [comment]
Okay, I haven't gotten around to taking a photo, so I'll do the lazy thing and use Google. Here's a screenshot from [Street View](https://www.google.com/maps/@42.3348794,-72.6685979,3a,15y,7.91h,76.27t/data=!3m6!1e1!3m4!1sLPtvOttA1IfbdYam79aVPA!2e0!7i16384!8i8192). (© Google, fair use). It's near the other images above.

![image](https://user-images.githubusercontent.com/6563664/136120851-99081784-9a15-4a4c-91ec-04751977d578.png)

I think the form does not give clear instructions about how to answer this case: "curb ramp" or "same level as the road"?

Based on your earlier comment here, I currently think it should be tagged as "same level", but before that, based on the images, I was tagging it as a ramp. After all, it ends in a ramp.

In other words, the form needs to clarify, what is the distinguishing feature of a curb ramp? Is it any ramp, or only one that is perpendicular to the sidewalk?

**Reply by mnalis** [reply]
According to wiki for [key:kerb](https://wiki.openstreetmap.org/wiki/Key:kerb) the main difference between `kerb=lowered` (_"curb ramp"_ and _"a bit higher than the road surface"_ answers in SC) and `kerb=flush` (_"same level as the road surface"_ SC answer) is that in the latter **_"level transition is unlikely to be apparent, to BLIND users, by tactile sensation"_**.

In my experience (I'm no expert by any stretch of imagination, but I do volunteer with blind user group in Croatia from time to time), blind users would quite likely **not have a problem detecting this slope change** when going from left to right, or right to left in that picture.

Hence, when moving horizontally in that picture, my answer would be `kerb=lowered`.  Sure, it's not a same degree of slope of `lowered` as in [this example in lower right](https://www.mapillary.com/app/?lat=45.804974008333&lng=15.9582636225&z=17&pKey=936980433731707&focus=photo&x=0.48885759641848436&y=0.5704592240780537&zoom=0) (which would be more of **"OMG I'm going to fall down!"** moment even for seeing person which was distracted by looking at their phone when they encountered it) -- but still detectable.

And when going up-to-down (or down-to-up) on that picture, it would be either `kerb=lowered` if the crossing was happening at about center of the screen, or `kerb=raised` (since we don't have `kerb=standard_height`) if the crossing was happening more at the left side of the picture.

**Reply by smichel17** [reply]
Thanks for pulling out the relevant sections, I was having trouble parsing that page (at least with the amount of attention I wanted to give it :P).

So, in other words, the determining factor is the **steepness** of the slope, not its **orientation.**

Based on that, I would be inclined to map the two pictures I originally posted as a ramp, not flush. Although those ramps are not perpendicular to the sidewalk, they are steep enough that a blind (or distracted) person would be surprised by the drop. And although they do not end exactly at the road, they are close enough that the purpose is clearly avoiding a raised curb.

**Comment by matkoniecz** [comment]
I am not sure is there agreement here among mappers how it should be mapped.

**Reply by smichel17** [reply]
Perhaps the quests should not be enabled until there is agreement? Based on the `🚧 Established tags only` [guideline](https://github.com/streetcomplete/StreetComplete/wiki/Adding-new-Quests-to-StreetComplete#general-guidelines)

**Reply by matkoniecz** [reply]
Tags are established, just exact boundary may be unclear. I suspect that it is muddled with different people tagging differently.

**Comment by westnordost** [comment]
Obviously, there is a smooth transition between "flat" and "ramp". Where this point is drawn is not clearly defined.

From my point of view, this is "flat" because I'd classify a ramp already as a normal kerb that has been cut in something like a 20°-45° angle while the image above is something like a 5° angle? For wheelchair users, there is very little difference between the image you posted and a completely flat kerb (i.e. the sidewalk would not be raised at all).

But in any case, where exactly to draw the line needs to be discussed on the OSM wiki. After that, we may want to adapt the pictures and/or add a short description.

**Comment by smichel17** [comment]
Okay. I'm converting this issue to a discussion since it is not currently actionable; perhaps I should have done this to begin with.
